packages/workspace/src/browser/workspace-service.ts

```diff
@@ -353,19 +426,40 @@ export class WorkspaceService implements FrontendApplicationContribution {
      */
     async containsSome(paths: string[]): Promise<boolean> {
         await this.roots;
-        if (this._workspaceFolder) {
-            const uri = new URI(this._workspaceFolder.uri);
-            for (const path of paths) {
-                const fileUri = uri.resolve(path).toString();
-                const exists = await this.fileSystem.exists(fileUri);
-                if (exists) {
-                    return exists;
+        if (this.opened) {
+            for (const root of this._roots) {
+                const uri = new URI(root.uri);
+                for (const path of paths) {
+                    const fileUri = uri.resolve(path).toString();
+                    const exists = await this.fileSystem.exists(fileUri);
+                    if (exists) {
+                        return exists;
+                    }
                 }
             }
         }
         return false;
     }
 
+    async saveWorkspaceMetaData(uri: URI) {
+        const uriStr = uri.withoutScheme().toString();
+        if (!await this.fileSystem.exists(uriStr)) {
+            await this.fileSystem.createFile(uriStr);
+        }
+        this.writeRootFolderConfigFile(uriStr, await this.roots);
```

Comment:
Why don't provide a content on `createFile` if it does not exist?

Reply:
`writeWorkspaceFile()` transforms the paths before performing the write operation, if I do `filesystem.createFile(uriStr, { content } )`, I would need to duplicate the paths transform logic. So, in this case, i guess leaving all the "file-writing logic" in one function (i.e., `writeWorkspaceFile()`) makes the code shorter and easier to follow.